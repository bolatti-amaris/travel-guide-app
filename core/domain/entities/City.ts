export interface City {
    id: number;
    key: string;
    name: string;
    nativeName: string;
    currency: string;
    language: string;
  }