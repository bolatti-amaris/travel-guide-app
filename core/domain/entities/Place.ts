export interface Place {
  type: 'monument' | 'restaurant';
  name: string;
  coordinates: [number, number];
}
