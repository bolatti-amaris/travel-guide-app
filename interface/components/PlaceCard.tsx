import React from 'react';
import { View, Text } from 'react-native';
import { Place } from '@/core/domain/entities/Place';

interface PlaceCardProps {
  place: Place;
}

const PlaceCard = ({ place }: PlaceCardProps) => {
  return (
    <View
      testID="place-card"
      className="bg-card-background dark:bg-card-background-dark p-4 rounded-sm mb-3"
    >
      <Text
        testID="place-name"
        className="text-xl font-work-bold text-text-header dark:text-text-header-dark mr-2"
      >
        {place.name}
      </Text>
      <Text
        testID="place-type"
        className="text-xl font-work-medium text-icon dark:text-icon-dark italic"
      >
        {place.type}
      </Text>
      <Text
        testID="place-coordinates"
        className="text-xs text-icon dark:text-icon-dark mt-1"
      >
        Lat: {place.coordinates[0]}, Lon: {place.coordinates[1]}
      </Text>
    </View>
  );
};

export default PlaceCard;
