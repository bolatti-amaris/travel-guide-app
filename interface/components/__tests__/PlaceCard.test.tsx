import React from 'react';
import { render } from '@testing-library/react-native';
import PlaceCard from '../PlaceCard';
import { Place } from '@/core/domain/entities/Place';

const mockPlace: Place = {
  name: 'Eiffel Tower',
  type: 'monument',
  coordinates: [48.8584, 2.2945],
};

describe('PlaceCard', () => {
  it('renders place data correctly', () => {
    const { getByTestId } = render(<PlaceCard place={mockPlace} />);

    expect(getByTestId('place-name').props.children).toBe('Eiffel Tower');
    expect(getByTestId('place-type').props.children).toBe('monument');

    const coordsText = getByTestId('place-coordinates').props.children.join('');
    expect(coordsText).toContain('48.8584');
    expect(coordsText).toContain('2.2945');
  });

  it('applies correct Tailwind classes', () => {
    const { getByTestId } = render(<PlaceCard place={mockPlace} />);

    expect(getByTestId('place-card').props.className).toContain(
      'bg-card-background',
    );
    expect(getByTestId('place-card').props.className).toContain(
      'dark:bg-card-background-dark',
    );
    expect(getByTestId('place-card').props.className).toContain('p-4');

    expect(getByTestId('place-name').props.className).toContain('text-xl');
    expect(getByTestId('place-name').props.className).toContain(
      'font-work-bold',
    );
    expect(getByTestId('place-name').props.className).toContain(
      'text-text-header',
    );
    expect(getByTestId('place-name').props.className).toContain(
      'dark:text-text-header-dark',
    );

    expect(getByTestId('place-type').props.className).toContain(
      'font-work-medium',
    );
    expect(getByTestId('place-type').props.className).toContain('italic');
    expect(getByTestId('place-type').props.className).toContain('text-icon');

    expect(getByTestId('place-coordinates').props.className).toContain(
      'text-xs',
    );
    expect(getByTestId('place-coordinates').props.className).toContain('mt-1');
  });

  it('matches snapshot', () => {
    const tree = render(<PlaceCard place={mockPlace} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
