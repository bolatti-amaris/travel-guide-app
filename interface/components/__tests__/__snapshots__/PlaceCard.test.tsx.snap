// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard matches snapshot 1`] = `
<View
  className="bg-card-background dark:bg-card-background-dark p-4 rounded-sm mb-3"
  testID="place-card"
>
  <Text
    className="text-xl font-work-bold text-text-header dark:text-text-header-dark mr-2"
    testID="place-name"
  >
    Eiffel Tower
  </Text>
  <Text
    className="text-xl font-work-medium text-icon dark:text-icon-dark italic"
    testID="place-type"
  >
    monument
  </Text>
  <Text
    className="text-xs text-icon dark:text-icon-dark mt-1"
    testID="place-coordinates"
  >
    Lat: 
    48.8584
    , Lon: 
    2.2945
  </Text>
</View>
`;
